tutorials/nservicebus-sagas/1-saga-basics/Solution/ClientUI/Worker.cs

```diff
@@ -0,0 +1,42 @@
+using Messages;
+
+namespace ClientUI;
+
+public class Worker(IMessageSession messageSession,ILogger<Worker> logger) : BackgroundService
+{
+    
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
```

Comment:
I think we can't use the hosted service in this way. The `ExecuteAsync` is called when the host is started and the host expects this method to bootstrap the service and return a `Task` representing the long-running part of the service.

I think that running the console loop in `Program.cs` might be an easier option here.